numpy/core/tests/test_arrayprint.py

```diff
@@ -395,6 +395,81 @@ def test_wide_element(self):
             "[ 'xxxxx']"
         )
 
+    def test_multiline_repr(self):
+        class MultiLine:
+            def __repr__(self):
+                return "Line 1\nLine 2"
+
+        a = np.array([[None, MultiLine()], [MultiLine(), None]])
+
+        assert_equal(
+            np.array2string(a),
+            '[[None Line 1\n'
+            '       Line 2]\n'
+            ' [Line 1\n'
+            '  Line 2 None]]'
+        )
+        assert_equal(
+            np.array2string(a, max_line_width=5),
+            '[[None\n'
+            '  Line 1\n'
+            '  Line 2]\n'
+            ' [Line 1\n'
+            '  Line 2\n'
+            '  None]]'
+        )
+        assert_equal(
+            repr(a),
+            'array([[None, Line 1\n'
+            '              Line 2],\n'
+            '       [Line 1\n'
+            '        Line 2, None]], dtype=object)'
+        )
+
+        class MultiLineLong:
+            def __repr__(self):
+                return "Line 1\nLooooooooooongestLine2\nLongerLine 3"
+
+        a = np.array([[None, MultiLineLong()], [MultiLineLong(), None]])
+        assert_equal(
+            repr(a),
+            'array([[None, Line 1\n'
+            '              LooooooooooongestLine2\n'
+            '              LongerLine 3          ],\n'
```

Comment:
Not sure what I think of this suffix padding, let's see whether other reviewers are in favor

Reply:
The suffix padding doesn't conform to any sort of style I'm familiar with, so I'm inclined to be against it. I don't think it does much to improve clarity (at least not in this case). Just my two cents!

Reply:
Looking again, it's sort of consistent with how we pad floats within a column with spaces to make them all the same width - so I'm +0.5 on keeping it.

Reply:
I believe in this case it's more clear that first element has multiline representation:
```
array([Line 1
       LooooooooooongestLine2
       LongerLine 3          , None], dtype=object)
```
However I agree that this looks neater:
```
array([Line 1
       LooooooooooongestLine2
       LongerLine 3, None], dtype=object)
```

Reply:
Don't let my stated opinion be a blocker - either one is okay with me, especially given the consistency argument.